syntax = "proto3";
package other_social;
import "google/protobuf/empty.proto";


message DemoMessage {
  int32 a = 1;
  string b = 2;
}
message SubSubSubNestedMessage{
  repeated DemoMessage repeated_demo = 1;
}
message SubSubNestedMessage{
  map<string, SubSubSubNestedMessage> map_demo = 1;
}
message SubNestedMessage{
  repeated SubSubNestedMessage repeated_demo = 1;
}

message NestedMessage{
  map<string, SubNestedMessage> map_demo = 1;
}


service Other {
  // pait: {"http_method": "POST", "url": "/other/nested-demo"}
  // pait: {"response_message": {"nested": "/map_demo/${}/repeated_demo/$[]/$.map_demo/${}/repeated_demo"}}
  rpc nested_demo (google.protobuf.Empty) returns (NestedMessage);
}
